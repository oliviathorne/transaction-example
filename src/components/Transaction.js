// @flow
import React from 'react';
import styled from 'styled-components';

const TransactionRow = styled.div`
  display: flex;
  border-bottom: 1px solid #edeef7;
  align-items: center;
  padding: 1rem 0;
`;

const TransactionColumn = styled.div`
  width: ${(props) => `${props.width}%`};
`;

export const Paragraph = styled.p`
  margin: 0;
  float: ${(props) => props.float || 'none'};
  font-size: ${(props) => (props.smallText ? '11pt' : 'inherit')};
  color: ${(props) => (props.smallText ? '#535353' : 'inherit')};
  margin-top: ${(props) => (props.smallText ? '.4rem' : 0)};
  font-weight: ${(props) => (props.bold ? '600' : 'inherit')};
`;

type Props = {
  name: { value: string },
  email: { value: string },
  amount: { value: string }
};

const Transaction = (props: Props) => {
  const AMOUNT = parseFloat(props.amount.value)
    .toFixed(2)
    .replace(/\B(?=(\d{3})+(?!\d))/g, ',');

  return (
    <TransactionRow>
      <TransactionColumn width={70}>
        <Paragraph>{props.name.value}</Paragraph>
        <Paragraph smallText>{props.email.value}</Paragraph>
      </TransactionColumn>
      <TransactionColumn width={30}>
        <Paragraph float="right" bold>
          Â£{AMOUNT}
        </Paragraph>
      </TransactionColumn>
    </TransactionRow>
  );
};

export default Transaction;
