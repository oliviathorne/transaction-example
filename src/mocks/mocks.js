export const mockState = {
  form: {
    fields: {},
    savedTransactions: [
      {
        name: {
          value: 'Test name'
        },
        email: {
          value: 'email@test.com'
        },
        amount: {
          value: '3000'
        }
      }
    ],
    startingBalance: 18000,
    totalAvailable: 18000,
    totalSent: 0
  }
};
