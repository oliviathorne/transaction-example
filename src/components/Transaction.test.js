import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mockState } from './../mocks/mocks';
import Transaction, { Paragraph } from './Transaction';

const mockStore = configureStore();
let store;
let wrapper;
const props = {
  name: {
    value: 'Test name'
  },
  email: {
    value: 'test@email.com'
  },
  amount: {
    value: '3000'
  }
};

beforeEach(() => {
  store = mockStore(mockState);
  wrapper = mount(
    <Provider store={store}>
      <Transaction {...props} />
    </Provider>
  );
});

describe('Transaction component', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders three paragraphs with the name, email and amount', () => {
    const paragraphs = wrapper.find(Paragraph);
    expect(paragraphs.length).toBe(3);

    paragraphs.forEach((paragraph, index) => {
      switch (index) {
        case 0:
          expect(paragraph.props().children).toEqual('Test name');
          break;
        case 1:
          expect(paragraph.props().children).toEqual('test@email.com');
          break;
        case 2:
          expect(paragraph.props().children).toEqual(['Â£', '3,000.00']);
          break;
        default:
          return;
      }
    });
  });
});
