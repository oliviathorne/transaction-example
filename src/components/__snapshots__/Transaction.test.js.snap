// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Transaction
    amount={
      Object {
        "value": "3000",
      }
    }
    email={
      Object {
        "value": "test@email.com",
      }
    }
    name={
      Object {
        "value": "Test name",
      }
    }
  >
    <styled.div>
      <StyledComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "jwZnsa",
              "rules": Array [
                "
  display: flex;
  border-bottom: 1px solid #edeef7;
  align-items: center;
  padding: 1rem 0;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa jwZnsa"
        >
          <styled.div
            width={70}
          >
            <StyledComponent
              forwardedClass={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "hTyRPc",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width={70}
            >
              <div
                className="sc-bwzfXH cvZEdJ"
                width={70}
              >
                <styled.p>
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jBFklx",
                          "rules": Array [
                            "
  margin: 0;
  float: ",
                            [Function],
                            ";
  font-size: ",
                            [Function],
                            ";
  color: ",
                            [Function],
                            ";
  margin-top: ",
                            [Function],
                            ";
  font-weight: ",
                            [Function],
                            ";
",
                          ],
                        },
                        "displayName": "styled.p",
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <p
                      className="sc-htpNat bwlXXh"
                    >
                      Test name
                    </p>
                  </StyledComponent>
                </styled.p>
                <styled.p
                  smallText={true}
                >
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jBFklx",
                          "rules": Array [
                            "
  margin: 0;
  float: ",
                            [Function],
                            ";
  font-size: ",
                            [Function],
                            ";
  color: ",
                            [Function],
                            ";
  margin-top: ",
                            [Function],
                            ";
  font-weight: ",
                            [Function],
                            ";
",
                          ],
                        },
                        "displayName": "styled.p",
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    smallText={true}
                  >
                    <p
                      className="sc-htpNat cRIDSO"
                    >
                      test@email.com
                    </p>
                  </StyledComponent>
                </styled.p>
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div
            width={30}
          >
            <StyledComponent
              forwardedClass={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "hTyRPc",
                    "rules": Array [
                      "
  width: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width={30}
            >
              <div
                className="sc-bwzfXH hTyRPc"
                width={30}
              >
                <styled.p
                  bold={true}
                  float="right"
                >
                  <StyledComponent
                    bold={true}
                    float="right"
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jBFklx",
                          "rules": Array [
                            "
  margin: 0;
  float: ",
                            [Function],
                            ";
  font-size: ",
                            [Function],
                            ";
  color: ",
                            [Function],
                            ";
  margin-top: ",
                            [Function],
                            ";
  font-weight: ",
                            [Function],
                            ";
",
                          ],
                        },
                        "displayName": "styled.p",
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <p
                      className="sc-htpNat jBFklx"
                    >
                      Â£
                      3,000.00
                    </p>
                  </StyledComponent>
                </styled.p>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Transaction>
</Provider>
`;
